crates/ibc/src/clients/ics07_tendermint/client_state.rs

```diff
@@ -253,6 +254,83 @@ impl ClientState {
             _ => Ok(()),
         }
     }
+
+    fn check_trusted_header(
+        consensus_state: &TmConsensusState,
+        header: &Header,
+    ) -> Result<(), Ics02Error> {
+        let trusted_val_hash = header.trusted_validator_set.hash();
+
+        if consensus_state.next_validators_hash != trusted_val_hash {
+            return Err(Error::misbehaviour_trusted_validator_hash_mismatch(
+                header.trusted_validator_set.clone(),
+                consensus_state.next_validators_hash,
+                trusted_val_hash,
+            )
+            .into());
+        }
+
+        Ok(())
+    }
+
+    fn check_misbehaviour_header(
```

Comment:
Maybe we don't need `misbehaviour` in the name here?
```suggestion
    fn check_header_and_validator_set(
```

...

Reply:
https://github.com/cosmos/ibc-rs/pull/215/commits/0ab1df91fe852644798e4e61ce08cc2188695e9a